### PR TITLE
Poll request

### LINKED ISSUE #426: Provider program to find new coming request
Polling process to query latest request id and compare with old id if get new request id, program will spin new go routine to handle new request.

Implementation detail:
- [ ] Setup initial state when program open first time
- [ ] Create new `BandStatefulClient`
- [ ] Loop to get new latest request id
- [ ] If latest request changed this program can spin new go routine to handle new request with requestId and pointer of bandStatefulClient

### PR DESCRIPTION
**Pull request by taobun**
fixed #426

### CODE REVIEW

File: chain/cmd/provider/main.go [hunk 1]
```diff
@@ -22,34 +22,78 @@ import (
...
 // TODO: Replace by `BandStatefulClient` after implementation finished.
 var bandProvider d3nlib.BandProvider
 var allowedCommands = map[string]bool{"curl": true, "date": true}
 
-func main() {
-	// Get environment variable
-	privS, ok := os.LookupEnv("PRIVATE_KEY")
-	if !ok {
-		log.Fatal("Missing private key")
+func getEnv(key, def string) string {
```
Comment (sorawit) [inline chain/cmd/provider/main.go]: let's spell it out `default`. I was confused at first with this `def` thinking it's like Python's `def` lol

File: chain/cmd/provider/main.go [hunk 2]
```diff
@@ -22,34 +22,78 @@ import (
...
+		newRequest, err := getLatestRequest()
+		if err != nil {
+			log.Println("Cannot get request number error: ", err.Error())
+		}
+
+		for currentRequest < newRequest {
+			currentRequest++
+			go newHandleRequest(currentRequest)
+		}
+		time.Sleep(2 * time.Second)
```
Comment (sorawit) [inline chain/cmd/provider/main.go]: Let's go lower. 1 second!

File: chain/cmd/provider/main.go [hunk 3]
```diff
@@ -22,34 +22,78 @@ import (
...
+	return tmp
+}
+
+var (
+	nodeURI  = getEnv("NODE_URI", "http://localhost:26657")
+	queryURI = getEnv("QUERY_URI", "http://localhost:1317")
+	privS    = getEnv("PRIVATE_KEY", "eedda7a96ad35758f2ffc404d6ccd7be913f149a530c70e95e2e3ee7a952a877")
+)
+
+func getLatestRequest() (uint64, error) {
```
Comment (sorawit) [inline chain/cmd/provider/main.go]: `getLatestRequestID`

File: chain/cmd/provider/main.go [hunk 4]
```diff
@@ -22,34 +22,78 @@ import (
...
 
 	var err error
-	bandProvider, err = d3nlib.NewBandProvider(priv)
+	bandProvider, err = d3nlib.NewBandProvider(nodeURI, priv)
 	if err != nil {
 		panic(err)
 	}
-	s := sub.NewSubscriber(viper.GetString("nodeURI"), "/websocket")
+
+	currentRequest, err := getLatestRequest()
```
Comment (sorawit) [inline chain/cmd/provider/main.go]: `currentRequestID`, `newRequestID`

File: chain/cmd/provider/main.go [hunk 5]
```diff
@@ -22,34 +22,78 @@ import (
...
+		}
+
+		for currentRequest < newRequest {
+			currentRequest++
+			go newHandleRequest(currentRequest)
+		}
+		time.Sleep(2 * time.Second)
+	}
+
+	s := sub.NewSubscriber(nodeURI, "/websocket")
```
Comment (sorawit) [inline chain/cmd/provider/main.go:96]: Let's remove all this unreachable code and the whole sub folder?

**Review by sorawit — CHANGES_REQUESTED** [review]

File: chain/cmd/provider/main.go (hunk 5 above)
Comment (taobun) [reply chain/cmd/provider/main.go:96]: We haven't finished new request handler, after implement those ticket. It will be removed later

**Comment by taobun** [comment]
Please review @sorawit @prin-r

**Review by sorawit — APPROVED** [review]

**Review by evilpeach — APPROVED** [review]
LGTM

**Pull request merged**